src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -624,6 +629,9 @@ int SolverInterfaceImpl:: setMeshVertex
   const double* position )
 {
   TRACE(meshID);
+  if(!_canSetMesh) {
+      ERROR("Cannot set the mesh in the current state!");
+  }
   Eigen::VectorXd internalPosition(_dimensions);
```

Comment:
This (and all other cases) should be replaced by a `CHECK`. 
Also, the error message needs to be more informative. Currently, I guess that users might be puzzled.

Reply:
A perfect error message is no easy task, though

Reply:
Instead of the meshID, we should give the mesh name as feedback to the user.

Reply:
I changed the `ERROR` to `CHECK` but had to do a bit more work to print the name of the mesh.

Problem is that the SolverInterface(Impl) provides neither a `hasMesh(int meshID)` nor a `getMeshName(int meshID)`.

To reduce clutter, I added a type MeshInfo which prints a given meshID and its name if the id is actually known by the SolverInterface. The factory function `SolverInterfaceImpl::makeMeshInfo(int meshID)` builds and returns an object of this type tyed to the SolverInterfaceImpl.
The `operator<<` is defined, so it can be inlined into `CHECK`/`ERROR`.


Example output of `cout << makeMeshInfo(meshID)`:
```
1:"MeshOne"
2:<unknown>
```

Reply:
I am not sure if I understand this. Which case is `<unknown>` here? 
Normally, you access the mesh via the participant `_accessor`:
```
_accessor->meshContext(meshID)
```
I have the feeling you are over-complicating things here. Sth like `makeMeshInfo` should not be part of `SolverInterfaceImpl` in my opinion.

Reply:
`<unknown>` is a mesh where the given `meshID` is unknown to the participant aka wrong input.

It is a small utility type to prevent additional control flow around the `ERROR` and  `CHECK` macros to cover the case that the meshID doesn't even exist.